bin/generate.py

```diff
@@ -835,57 +818,32 @@ def check_deterministic(force=False):
                         f'All values in [{t.seed}, {new_seed}] give the same result.',
                     )
 
-        def add_testdata_to_cache():
-            # Used to identify generated testcases
-            generator_config.hashed_in.add(hash_file_content(infile))
-
-            # Store the generated testdata for deduplication test cases.
-            hashes = {}
+        def generate_in():
+            nonlocal meta_yaml
 
-            # remove files that should not be considered for this testcase
-            extensions = config.KNOWN_TESTCASE_EXTENSIONS.copy()
-            if t.root not in config.INVALID_CASE_DIRECTORIES[1:]:
-                extensions.remove('.ans')
-            if t.root not in config.INVALID_CASE_DIRECTORIES[2:]:
-                extensions.remove('.out')
-
-            for ext in extensions:
-                if target_infile.with_suffix(ext).is_file():
-                    hashes[ext] = hash_file(target_infile.with_suffix(ext))
-
-            # build ordered list of hashes we want to consider
-            test_hash = [hashes[ext] for ext in extensions if ext in hashes]
-
-            # combine hashes
-            if len(test_hash) == 1:
-                test_hash = test_hash[0]
-            else:
-                test_hash = combine_hashes(test_hash)
-
-            # check for duplicates
-            if test_hash not in generator_config.generated_testdata:
-                generator_config.generated_testdata[test_hash] = t
-            else:
-                bar.warn(
-                    f'Testcase {t.path} is equal to {generator_config.generated_testdata[test_hash].path}.'
-                )
+            # create expected cache entry for generate
+            generate_hashes = dict()
+            if t.copy:
+                generate_hashes['source_hash'] = t.hash
+            for ext, string in t.hardcoded.items():
```

Comment:
Maybe it's actually cleaner in the end to only check the `.in` here, and make a separate `generate_meta` or so that checks the `.desc`, `.hint`, and image extensions?